tensorflow/core/common_runtime/eager/kernel_and_device.cc

```diff
@@ -278,6 +280,47 @@ Status KernelAndDeviceOp::Run(
 
   OpKernelContext context(&params);
 
+  string msg;
+  if (nvtx::NvtxRangesEnabled() || nvtx::NvtxRangesDetailedEnabled()) {
+    if (nvtx::NvtxRangesDetailedEnabled()) {
+      std::vector<string> args_pieces;
+      for (int i = 0; i < inputs.GetTensorValues()->size(); i++) {
+        if (i == 10) {
+          // Truncate long arg lists and indicate with an ending null value.
+          args_pieces.push_back("null");
+          break;
+        }
+        const auto& shape = inputs.GetTensorValues()->at(i).tensor->shape();
+        string shape_str = shape.unknown_rank() ? "null" : shape.DebugString();
+        args_pieces.push_back(strings::StrCat("{\"name\":\"",
+                                              kernel_->def().input(i),
+                                              "\",\"shape\":", shape_str, "}"));
+      }
+      std::vector<string> attrs_pieces;
+      const auto& attrs = kernel_->def().attr();
+      for (auto it = attrs.begin(); it != attrs.end(); ++it) {
+        const string& key = it->first;
+        const AttrValue& value = it->second;
+        // Exclude types that aren't useful for profiling.
+        if (value.value_case() == AttrValue::kFunc ||
+            value.value_case() == AttrValue::kPlaceholder ||
+            value.value_case() == AttrValue::VALUE_NOT_SET) {
```

Comment:
what if AttrValue::kTensor is really big here?

Reply:
Changed code to avoid deserializing TensorProto if larger than 128K elements. Only three elements are ever included in the NVTX tag per Tensor::DebugString().